### PR TITLE
implement `txs.Visitor` in X chain wallet

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Resolves a TODO.

## How this works

Implements `txs.Visitor` interface

## How this was tested

CI

### CODE REVIEW

**Review by coffeeavax — APPROVED** [review]

File: wallet/chain/x/backend.go [hunk 1]
```diff
@@ -4,8 +4,6 @@
 package x
 
 import (
-	"fmt"
-
 	stdcontext "context"
```
Comment (danlaine) [inline wallet/chain/x/backend.go:7]: @StephenButtolph Why is this aliased in wallet files?

File: wallet/chain/x/backend.go [hunk 2]
```diff
@@ -48,38 +46,15 @@ func NewBackend(ctx Context, chainID ids.ID, utxos ChainUTXOs) Backend {
...
-				return err
-			}
-		}
-	default:
-		return fmt.Errorf("%w: %T", errUnknownTxType, tx.Unsigned)
+	txID := tx.ID()
+	err := tx.Unsigned.Visit(&backendVisitor{
+		b:    b,
+		ctx:  ctx,
+		txID: txID,
```
Comment (danlaine) [inline wallet/chain/x/backend.go:54]:
```suggestion
	err := tx.Unsigned.Visit(&backendVisitor{
		b:    b,
		ctx:  ctx,
		txID: tx.ID(),
```

File: wallet/chain/x/backend.go (hunk 1 above)
Comment (dhrubabasu) [reply wallet/chain/x/backend.go:7]: We have a `context` already: https://github.com/ava-labs/avalanchego/blob/master/wallet/chain/x/context.go#L24

**Review by danlaine — APPROVED** [review]
1 nit otherwise lgtm

File: wallet/chain/x/backend.go (hunk 1 above)
Comment (danlaine) [reply wallet/chain/x/backend.go:7]: ty

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**